### PR TITLE
[SOAR-8721] Google Directory - Added new action Get User's Contact Information

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Added new action Get User's Contact Information

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### Run

<details>

```
{
  "body": {
    "log": "Using OAuth scope (['https://www.googleapis.com/auth/admin.directory.user']) for connections\nImpersonating user admin@komandops.org\nrapid7/Google Directory:2.1.0. Step name: get_user_contact\n",
    "meta": {},
    "output": {
      "contact": {
        "addresses": [
          {
            "street": "Test"
          }
        ],
        "emails": [
          "test@example.com",
          "test2@domain.com"
        ],
        "phones": [
          "111111111"
        ]
      }
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}
```

<summary>
docker run --rm -i rapid7/google_directory:2.1.0 run < tests/get_user_contact.json
</summary>
</details>

### Test

<details>

```
{
  "body": {
    "log": "Using OAuth scope (['https://www.googleapis.com/auth/admin.directory.user']) for connections\nImpersonating user test@domain.com\nrapid7/Google Directory:2.1.0. Step name: get_user_contact\n",
    "status": "ok",
    "meta": {},
    "output": {
      "success": true
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

<summary>
docker run --rm -i rapid7/google_directory:2.1.0 run < tests/get_user_contact.json
</summary>
</details>

### CODE REVIEW

File: plugins/google_directory/icon_google_directory/util/tools.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Message:
+    USER_CONTACT_CAUSE = "Contact informations were not found in server's response."
```
Comment (cmcnally-r7) [inline plugins/google_directory/icon_google_directory/util/tools.py:2]:
```suggestion
    USER_CONTACT_CAUSE = "Contact information was not found in server's response."
```

File: plugins/google_directory/icon_google_directory/util/tools.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    USER_CONTACT_ASSISTANCE = "Please check and verify if user exists."
```
Comment (cmcnally-r7) [inline plugins/google_directory/icon_google_directory/util/tools.py:3]:
```suggestion
    USER_CONTACT_ASSISTANCE = "Please check and verify if the user exists."
```

File: plugins/google_directory/plugin.spec.yaml [hunk 3]
```diff
@@ -33,6 +33,38 @@ types:
       description: Full name of the user
       type: string
       required: false
+  address:
+    country:
+      title: Country
+      description: Country
```
Comment (cmcnally-r7) [inline plugins/google_directory/plugin.spec.yaml:39]:
```suggestion
      description: User's nation/territory
```

File: plugins/google_directory/plugin.spec.yaml [hunk 4]
```diff
@@ -33,6 +33,38 @@ types:
...
+      type: string
+      required: false
+    street:
+      title: Street
+      description: User's street address
+      type: string
+      required: false
+    postal_code:
+      title: Postal Code
+      description: User's ZIP or postal code, if applicable
```
Comment (cmcnally-r7) [inline plugins/google_directory/plugin.spec.yaml:49]:
```suggestion
      description: User's ZIP or postal code
```

File: plugins/google_directory/plugin.spec.yaml [hunk 5]
```diff
@@ -125,6 +157,22 @@ actions:
         description: Users in the domain
         required: true
         type: '[]user'
+  get_user_contact:
+    title: Get User's Contact Information
+    description: Retrieves all contact information from a specific user in domain by email
```
Comment (cmcnally-r7) [inline plugins/google_directory/plugin.spec.yaml:162]:
```suggestion
    description: Retrieves all contact information from a specific user in the domain by email
```

File: plugins/google_directory/unit_test/responses/suspend_user.json.resp [hunk 6]
```diff
@@ -1,4 +1,4 @@
-{
+ {
```
Comment (cmcnally-r7) [inline plugins/google_directory/unit_test/responses/suspend_user.json.resp:1]: Apologies if this is pedantic. 
Will this still work with the extra space here?

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]

File: plugins/google_directory/unit_test/responses/suspend_user.json.resp (hunk 6 above)
Comment (igorski-r7) [reply plugins/google_directory/unit_test/responses/suspend_user.json.resp:1]: It worked but you're right that it didn't have to look this way 😃

**Review by cmcnally-r7 — APPROVED** [review]

**Review by ablakley-r7 — APPROVED** [review]
Looks good. Only thought would be transforming 'phones' and 'emails' to 'phone_numbers' and 'email_addresses' with the thought that if you use it in a sentence you'd say the latter.

**Review by mberezin-r7 — APPROVED** [review]

**Pull request merged**